pkg/user/user_integration_test.go

```diff
@@ -84,20 +76,102 @@ func TestUserHandler(t *testing.T) {
 		require.Equal(t, "user2@dhis2.org", user2.Email)
 		require.Empty(t, user2.Password)
 
-		u2, err := userService.FindById(context.Background(), user2.ID)
-		require.NoError(t, err)
-		err = userService.ValidateEmail(u2.EmailToken)
+		createUser(t, client, userService, "user3@dhis2.org", "oneoneoneoneoneoneone111")
+	}
+
+	t.Run("SignOut", func(t *testing.T) {
+		t.Parallel()
+
+		expires, err := time.Parse(time.RFC3339, "0001-01-01T00:00:00Z")
 		require.NoError(t, err)
 
-		var user3 model.User
-		client.PostJSON(t, "/users", strings.NewReader(`{
-			"email":    "user3@dhis2.org",
-			"password": "oneoneoneoneoneoneone111"
-		}`), &user3)
+		expectedCookies := []*http.Cookie{
+			{
+				Name:       "accessToken",
+				Value:      "",
+				Path:       "/",
+				Domain:     "",
+				Expires:    expires,
+				RawExpires: "",
+				MaxAge:     -1,
+				Secure:     true,
+				HttpOnly:   true,
+				SameSite:   0,
+				Raw:        "accessToken=; Path=/; Max-Age=0; HttpOnly; Secure",
+				Unparsed:   nil,
+			},
+			{
+				Name:       "refreshToken",
+				Value:      "",
+				Path:       "/",
+				Domain:     "",
+				Expires:    expires,
+				RawExpires: "",
+				MaxAge:     -1,
+				Secure:     true,
+				HttpOnly:   true,
+				SameSite:   0,
+				Raw:        "refreshToken=; Path=/; Max-Age=0; HttpOnly; Secure",
+				Unparsed:   nil,
+			},
+			{
+				Name:       "rememberMe",
+				Value:      "",
+				Path:       "/",
+				Domain:     "",
+				Expires:    expires,
+				RawExpires: "",
+				MaxAge:     -1,
+				Secure:     true,
+				HttpOnly:   true,
+				SameSite:   0,
+				Raw:        "rememberMe=; Path=/; Max-Age=0; HttpOnly; Secure",
+				Unparsed:   nil,
+			},
+		}
 
-		require.Equal(t, "user3@dhis2.org", user3.Email)
-		require.Empty(t, user3.Password)
-	}
+		t.Run("NoToken", func(t *testing.T) {
+			request := client.NewRequest(t, http.MethodDelete, "/users", nil)
+
+			response, err := client.Client.Do(request)
+			require.NoError(t, err)
+
+			require.Equal(t, http.StatusUnauthorized, response.StatusCode)
+			assert.EqualValues(t, expectedCookies, response.Cookies())
+		})
+
+		t.Run("ValidToken", func(t *testing.T) {
+			createUser(t, client, userService, "user4@dhis2.org", "oneoneoneoneoneoneone111")
+
+			var tokens *token.Tokens
+			client.PostJSON(t, "/tokens", strings.NewReader(`{}`), &tokens, inttest.WithBasicAuth("user4@dhis2.org", "oneoneoneoneoneoneone111"))
```

Comment:
is passing `strings.NewReader(`{}`)` necessary?
we did have working requests using no body so passing `nil` before

Reply:
The difference is that now the endpoint accepts an optional request body.

In the handler I can do the following

	var request signInRequest
	if err := handler.DataBinder(c, &request); err != nil && err != io.EOF {
		_ = c.Error(err)
		return
	}

The relevant part is the last condition which means we accept io.EOF errors. Can you think of a better solution?

Reply:
I've committed the above. Let me know if you think it's a bad idea :slightly_smiling_face:

Reply:
> The difference is that now the endpoint accepts an optional request body.
if the body is optional than passing `nil` in the test should work

I think the solution you used in the handler is alright for now. We might need to rethink the `DataBinder` at some point though 😄